csrc/transform_view.cpp

```diff
@@ -137,18 +138,23 @@ class ViewTransform : public Transform {
 
     auto root_domain_pos = std::distance(root_domain.begin(), root_domain_it);
 
-    bool is_expanded_dim = id->hasExpandedExtent();
+    auto iter_type = id->getIterType();
+    auto extent = id->extent();
+    Val* expanded_extent =
+        id->hasExpandedExtent() ? id->expandedExtent() : nullptr;
 
-    auto extent = is_expanded_dim ? id->expandedExtent() : id->extent();
-
-    auto cloned_id =
-        IterDomainBuilder(id)
-            .iter_type(
-                is_expanded_dim ? IterType::Iteration : id->getIterType())
-            .extent(extent)
-            .expanded_extent(nullptr)
-            .is_rfactor_domain(true)
-            .build();
+    if (realize_expand && id->hasExpandedExtent()) {
+      // Convert an expanded broadcast to Iteration (resolve the broadcast)
+      iter_type = IterType::Iteration;
+      extent = id->getMaybeExpandedExtent();
```

Comment:
```suggestion
      extent = expanded_extent;
```

Reply:
I prefer to leave this explicit for clarity since I'm changing the value of `expanded_extent` in the next line.

Reply:
Actually on second thought I like this suggestion better.